Mathlib/Analysis/InnerProductSpace/Projection.lean

```diff
@@ -845,6 +845,14 @@ theorem Submodule.isCompl_orthogonal_of_completeSpace [HasOrthogonalProjection K
   ⟨K.orthogonal_disjoint, codisjoint_iff.2 Submodule.sup_orthogonal_of_completeSpace⟩
 #align submodule.is_compl_orthogonal_of_complete_space Submodule.isCompl_orthogonal_of_completeSpace
 
+theorem eq_iff_orthogonalComplement_eq {L : Submodule 𝕜 E}[HasOrthogonalProjection K]
+    [HasOrthogonalProjection L] : K = L ↔ Kᗮ = Lᗮ := by
+   constructor
+   · exact fun a ↦ congrArg Submodule.orthogonal a
+   · intro H
+     rw [← (Submodule.orthogonal_orthogonal K), ← (Submodule.orthogonal_orthogonal) L]
+     exact congrArg Submodule.orthogonal H
```

Comment:
With the above suggested change, you can do this for the proof (otherwise you have to reverse the order of the arguments):
```
  ⟨fun h ↦ by simpa using congr(Submodule.orthogonal $(h)),
    fun h ↦ congr(Submodule.orthogonal $(h))⟩
```
A few notes about this proof:
1. `congr()` is a *really* cool way of building equalities from existing ones. `congrArg` (or `congr_arg`) is the way of doing it by hand, but after you try to prove `a + b = c + d` from `h₁ : a = c` and `h₂ : b = d` using `congr_arg` alone (not allowing yourself `rw`), you will realize the full power of `congr`, which can do this simply as `congr($(h₁) + $(h₂))`.
2. `simpa using <term>` is nice because it allows you to supply a term which only needs to agree with the goal up to simplification. So if Lean can simplify this term and the goal to the same thing, it closes the goal.
3. While I see what you were doing with the backwards rewriting, it can sort of be a sign (not always) that you're doing something in a non-optimal way if you are reverse-rewriting `simp` lemmas. At the very least, if you try to `simp` after you undo your work. So, instead, it can sometimes be helpful to build a *more* complicated term, and then simplify it, which is what I've done here. In the first line, `h : Kᗮ = Lᗮ`, then the `congr(Submodule.orthogonal $(h)) : Kᗮᗮ = Lᗮᗮ`, and the `simpa` uses `Submodule.orthogonal_orthogonal` under the hood to remove the doubled perps to yield `K = L`, as desired.

This last one is a pretty common trick, and I think it's closer to what you might do on paper.

Reply:
Thank you for the helpful comments!